### PR TITLE
fix: resolving property references in xml files

### PR DESCRIPTION
**Pull request by Spezialissimo**
# PROBLEM
Ivy does not replace property references in xml files. That means that when they are present, we might get a report that looks like this:
```
unrecognized,${properties.license.name} (https://example.org/${project.artifactId}),com.example # fake-lib # 1.0.0
```
I noticed that in some (rare) cases the original xml file only contains the url as info about the license. For this reason, I thought it would have been great if we were able to replace such references before creating the report.

# SOLUTION
The current implementation uses a best effort approach.
If it finds any `${}` it will try to find the xml file that ivy is retrieving information from. If it's able to find the file, it will try to find the replacement and update the UpdateReport object. If any of this fails, nothing happens and the references don't get replaced.

I'm not an expert of how the information gets originally retrieved, for this reason it will check, in order:
1. `.pom` file with same path and name than the `.jar` file.
2. any `.xml.original` file present two directories up from the `.jar` file.
3. any `.xml` file present two directories up from the `.jar` file.

As these are the cases I found in my experience. If you know of way to improve this search please let me know.

Also, I added a test to check that the replacement is working. I want to reiterate that i'm not an expert of the various edge cases so let me know if you think the tests can be improved.

Additionally, other cases of failures in creating the URI object for the homepage field are now handled, taking in consideration that the ".toUri" function seems to not disclose every kind of exception it can throw.

### CODE REVIEW

**Comment by Spezialissimo** [comment]
@mdedetrich Let me know if there is something else I should look into for this PR, otherwise it would be great if it could be merged and released

File: src/main/scala/sbtlicensereport/license/LicenseReport.scala [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
 package license
 
-import java.net.URISyntaxException
-
-import sbt._
+import java.net.{ MalformedURLException, URISyntaxException }
+import java.io.File
+import scala.util.control.NonFatal
+import scala.xml.{ Elem, XML }
+import sbt.*
```
[the remark this answers is not in the document]
Comment (Spezialissimo) [reply src/main/scala/sbtlicensereport/license/LicenseReport.scala]: Thank you, I've pushed the change

**Comment by Spezialissimo** [comment]
Thank you @mdedetrich and @raboof for taking a look into this!
